pkg/daemon/ceph/osd/remove.go

```diff
@@ -93,9 +94,10 @@ func removeOSD(context *clusterd.Context, clusterInfo *client.ClusterInfo, osdID
 		}
 		const pvcLabelName = "ceph.rook.io/pvc"
 		if pvcName, ok := deployment.GetLabels()[pvcLabelName]; ok {
-			prepareJobList, err := context.Clientset.BatchV1().Jobs(clusterInfo.Namespace).List(metav1.ListOptions{LabelSelector: pvcLabelName + pvcName})
+			labelSelector := fmt.Sprintf("%s=%s", pvcLabelName, pvcName)
```

Comment:
```suggestion
			labelSelector := fmt.Sprintf("%s=%s", osd.OSDOverPVCLabelKey, pvcName)
```

And remove the constant L95? We can leave without this if this makes the BP problematic though.